cmd/boulder-va/gsb.go

```diff
@@ -5,34 +5,109 @@ import (
 	"fmt"
 	"os"
 
+	safebrowsingv4 "github.com/google/safebrowsing"
 	"github.com/letsencrypt/boulder/cmd"
 	"github.com/letsencrypt/boulder/va"
 	safebrowsing "github.com/letsencrypt/go-safe-browsing-api"
 )
 
-// newGoogleSafeBrowsing returns nil if the GoogleSafeBrowsing struct given is
-// nil. If an empty Google API key or an unreadable data directory is in the
-// GoogleSafeBrowsing config struct, this function runs cmd.FailOnError.
-func newGoogleSafeBrowsing(gsb *cmd.GoogleSafeBrowsingConfig) va.SafeBrowsing {
+var (
+	NilConfigErr   = errors.New("Google Safe Browsing config was nil")
+	EmptyAPIKeyErr = errors.New("a Google Safe Browsing config was given but " +
+		"it did not include a Google API key in APIKey")
+	EmptyDataDirErr = errors.New("a Google Safe Browsing config was given but " +
+		"it did not include a DataDir for persistence")
+	MissingDataDirErr = errors.New("a Google Safe Browsing data directory was " +
+		"given but it does not exist")
+	BadDataDirErr = errors.New("a Google Safe Browsing data directory was " +
+		"given but it cannot be opened")
+)
+
+// configCheck returns an error if:
+// * the gsb config struct given is nil
+// * the gsb config struct's APIKey is empty
+// * the gsb config struct's DataDir is empty
+// * the gsb config struct's DataDir doesn't exist or isn't readable
+func configCheck(gsb *cmd.GoogleSafeBrowsingConfig) error {
 	if gsb == nil {
-		return nil
+		return NilConfigErr
 	}
 	if gsb.APIKey == "" {
-		cmd.FailOnError(errors.New(""), "a Google Safe Browsing config was given but it did not include a Google API key in APIKey")
+		return EmptyAPIKeyErr
 	}
 	if gsb.DataDir == "" {
-		cmd.FailOnError(errors.New(""), "a Google Safe Browsing config was given but it did not include data directory to store the hashes file in DataDir")
+		return EmptyDataDirErr
 	}
-
 	f, err := os.Open(gsb.DataDir)
+	// NOTE: Using `defer f.Close()` instead makes errcheck unhappy.
+	defer func() { _ = f.Close() }()
```

Comment:
This should be done after `err` is checked as there are paths that will cause the returned `*os.File` to be `nil` and `err` to be non-`nil`.